src/net/backend.py

```diff
@@ -27,25 +27,29 @@ def __init__(self, port: int) -> None:
         self.server = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
         self.server.bind(('0.0.0.0', port))
         self.server.listen()
+        self.server.settimeout(3)
 
     def receive(self) -> (IpAddress, str):
         # This can only receive from one peer at a time, hopefully that is fine
-        conn, source = self.server.accept()
-        data = bytearray()
-        while True:
-            frame = conn.recv(1000)
-            _logger.info(f'Received frame of {len(frame)} bytes')
-            if not frame:
-                break # Sender closed connection, which it will always do with our send(...)
-            data.extend(frame)
+        try:
+            conn, source = self.server.accept()
+            data = bytearray()
+            while True:
+                frame = conn.recv(1000)
+                _logger.info(f'Received frame of {len(frame)} bytes')
+                if not frame:
+                    break # Sender closed connection, which it will always do with our send(...)
+                data.extend(frame)
             
-        return (f'{source[0]}:{source[1]}', data.decode('utf-8'))
-    
+            return (f'{source[0]}:{source[1]}', data.decode('utf-8'))
+        except socket.timeout:
+            return (None, None)
+
     def send(self, dest: IpAddress, data: str) -> bool:
         # Connect, send, close - inefficient, but easy
         client = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
         client.settimeout(1)
-        _logger.info(f'Sending {len(data)} characters to {dest}')
+        _logger.info(f'Sending {len(data)} bytes to {dest}')
```

Comment:
It is actually characters, one character could theoretically be multiple bytes :)

Reply:
Then there should also be "characters" in line 39?